### PR TITLE
Create GC-05-18.md

### PR DESCRIPTION
**Pull request by tlively**
@rossberg, thanks for presenting! Can you upload your slides?

### CODE REVIEW

File: gc/2021/GC-05-18.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+RT: I understand it's complicated, asking if you have worked through to see if it works
+
+AR: no, not in that detail. I'm pretty confident that it works sufficiently well, iso-recursive is often used as the type theorteic model of nominal types
+
+RT: some red flags in representation. E.g. non-uniform recursion and how it is decidable. But it is known that non-uniform recursion that can handle OO subtyping, then it is undecidable
+
+CW: <something> bounded quantification
+
+RT: No, generics are undecidable with iso-recursive types without bounded quantification. Paper by <missed> names I presented to the group back in January.
```
Comment (tlively) [inline gc/2021/GC-05-18.md:124]: @RossTate, can you fill in the paper author here?

File: gc/2021/GC-05-18.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+AR: this might be actually a reason to allow nesting of recursion groups. Because you basically have recursion over recursive things. The subtyping rule for iso-rec is one rec group is subtype of another rec-group if you have both depth and width subtyping on components. Means that you can only extend one group in one dimension at a time. But if you have two mutually recursive classes, then the recursion group, for each clas the recursion between methods, then recursion between the two classe, which is one level up. The inner one want to be a subtype, but not on the combined thing.
+
+RT: I understand it's complicated, asking if you have worked through to see if it works
+
+AR: no, not in that detail. I'm pretty confident that it works sufficiently well, iso-recursive is often used as the type theorteic model of nominal types
+
+RT: some red flags in representation. E.g. non-uniform recursion and how it is decidable. But it is known that non-uniform recursion that can handle OO subtyping, then it is undecidable
+
+CW: <something> bounded quantification
```
Comment (tlively) [inline gc/2021/GC-05-18.md:122]:
```suggestion
CW: Undecidable in the presence of bounded quantification?
```
@conrad-watt, does this get the gist of what you were saying?

Comment (conrad-watt) [reply gc/2021/GC-05-18.md:122]: perfect!

File: gc/2021/GC-05-18.md (hunk 1 above)
Comment (RossTate) [reply gc/2021/GC-05-18.md:124]:
```suggestion
RT: No, generics are undecidable with iso-recursive types without bounded quantification. Follows from the [paper](https://kar.kent.ac.uk/58183/7/javats.pdf) by Radu Grigore as I presented to the group back in [December](https://github.com/WebAssembly/meetings/blob/main/gc/2020/GC-12-15.md).
```
Here you go! Though, looking back, apparently it was December and not January.

Comment (tlively) [reply gc/2021/GC-05-18.md]: Excellent, thanks!

**Comment by RossTate** [comment]
Should this be merged in? (I just noticed it while looking for yesterday's notes.)

**Comment by tlively** [comment]
Oops, yes, thanks!

**Pull request merged**